### PR TITLE
Create postgre script for creating tables

### PR DESCRIPTION
**Pull request by kimlisa**

### CODE REVIEW

File: sql/S1_CREATE_DOMAINS.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+CREATE DOMAIN user_name AS
+  VARCHAR(32) NOT NULL CHECK (VALUE ~ '^[[:alpha:]]+(([''.\s-][[:alpha:]\s])?[[:alpha:]]*)*$');
```
Comment (faraonc) [inline sql/S1_CREATE_DOMAINS.sql]: What is the purpose of `~`?

File: sql/S4_CREATE_SHARED_TABLE.sql [hunk 2]
```diff
@@ -0,0 +1,7 @@
+-- uuid and duid act as unique identifier b/c docs can be shared to any user
+CREATE TABLE shared_documents
+(
+  PRIMARY KEY (uuid, duid),
+  uuid ULID   REFERENCES user_account(uuid) ON DELETE CASCADE,
+  duid KSUID  REFERENCES documents(duid) ON DELETE CASCADE
+);
```
Comment (faraonc) [inline sql/S4_CREATE_SHARED_TABLE.sql]: You can combine all (S1-S4) SQL scripts as `<version>_initial_schema.sql`, where version depends on the DB migration or versioning that we will use. Read my comment below.

File: sql/test_create.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+docker rm -f devtest;\
+docker run --name devtest --mount type=bind,source="$(pwd)",target=/sql -d postgres:alpine;\
+sleep 5;\
+docker exec -it devtest psql -U postgres -c "CREATE DATABASE testdb;";\
+docker exec -it devtest psql -U postgres -f /sql/S1_CREATE_DOMAINS.sql;\
+docker exec -it devtest psql -U postgres -f /sql/S2_CREATE_USER_TABLE.sql;\
+docker exec -it devtest psql -U postgres -f /sql/S3_CREATE_DOCUMENT_TABLE.sql;\
+docker exec -it devtest psql -U postgres -f /sql/S4_CREATE_SHARED_TABLE.sql;\
```
Comment (faraonc) [inline sql/test_create.sh]: The SQL scripts in dev are the same sql scripts to be used in production, therefore DB management and versioning is important.

There are several ways of versioning sql scripts and DB migration. These are the examples:
 - https://bitbucket.org/liamstask/goose/
 - https://github.com/golang-migrate/migrate guide -> https://github.com/golang-migrate/migrate/blob/master/MIGRATIONS.md
- https://github.com/robinjoseph08/go-pg-migrations

I am open to any of these three.

File: sql/S1_CREATE_DOMAINS.sql (hunk 1 above)
Comment (kimlisa) [reply sql/S1_CREATE_DOMAINS.sql]: it is a regex operator that returns true if regex match, false otherwise

**Pull request closed without merging**